### PR TITLE
Deployment runbooks

### LINKED ISSUE #1912: Add instructions to deployment docs (and link from release app workflow inputs) for how to retrieve the SHA for deployment
## Problem

It isn't clear how to get the SHA to put as input for the release app workflow. Worse, it's different for different parts of the stack.

## Description

Update the deployment documentation to describe how to retrieve the correct SHA to use for each application.

Note that for frontend and API these can be retrieved from their respective version endpoints, but for catalog and ingestion server these need to be pulled from GHCR.

## Alternatives

If it's easy to pull the SHA directly from the CR or the version endpoints per app, do that, maybe as a Python script, and make the input optional for whenever we want to release the non-current staging version (for whatever reason).

### LINKED ISSUE #1981: Runbook for deployments
## Problem

We have excellent documentation explaining how deployments work; but we lack a simple, step-by-step guide for executing a deployment.

## Description

We could create a runbook in our documentation with a list of the steps to deploy our ECS services. This would be a general-purpose guide for any service. It would make sense for this runbook to live alongside the one outlined in #1923.

Some steps this would likely include:

- Check the version sha of staging
- Run the "Release app" workflow
- Approve the release note PR
- Click through the production application?
- Execute the "failed deploy" runbook, if a problem occurred

#  Alternatives

### PR DESCRIPTION
**Pull request by zackkrida**
## Fixes

Fixes #1912 
Fixes #1981

## Description

Adds a runbook for deploying each application. Please review the contents of each and let me know if I've missed anything, or if you feel the document structure I've used is appropriate.

## Testing Instructions

Review the generated docs preview.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [x] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: documentation/catalog/guides/deploy.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# Deployment runbook
+
+## Setup
+
+1. Check
+   [airflow](https://href.li/?https://airflow.openverse.engineering/home?tags=data_refresh)
```
Comment (dhruvkb) [inline documentation/catalog/guides/deploy.md:6]:
```suggestion
   [Airflow](https://href.li/?https://airflow.openverse.engineering/home?tags=data_refresh)
```

File: documentation/catalog/guides/deploy.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+   to make sure a data refresh isn't occurring.
+1. Record and verify the latest commit `sha` of the catalog in
```
Comment (dhruvkb) [inline documentation/catalog/guides/deploy.md:8]:
```suggestion
1. Record and verify the latest commit SHA of the catalog Docker image in
```

File: documentation/catalog/guides/deploy.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+   [the catalog package directory](https://github.com/wordpress/openverse/pkgs/container/openverse-catalog).
+1. Release the app via
+   [GitHub workflow](https://github.com/WordPress/openverse/actions/workflows/release-app.yml).
+   Click the "Run workflow" button, choose "catalog" from the dropdown, and
+   supply the `sha` identified in step 1.
```
Comment (dhruvkb) [inline documentation/catalog/guides/deploy.md:13]:
```suggestion
   supply the SHA identified in step 1.
```

File: documentation/catalog/guides/deploy.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+## Deployment
+
+1. After the app is built and tagged, deploy staging:
+   1. Checkout the
+      [infrastructure repository](https://github.com/wordpress/openverse-infrastructure)
+      and bump the ingestion server version with the `just bump dev catalog`
+      command.
+   1. `just apply dev catalog` and verify the plan before deploying.
+2. Deploy production:
+   1. Update the value of `data_refresh_cleared` to `true` in the
```
Comment (dhruvkb) [inline documentation/catalog/guides/deploy.md:24]: Missed the step to not commit this change.

File: documentation/api/guides/deploy.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+   and
+   [the api package directory](https://github.com/wordpress/openverse/pkgs/container/openverse-api).
+   Verify that the commit `sha` live on the staging site is also tagged with
+   `latest` in the package directory.
+
+   ```{figure} /_static/package_directory_example.png
+   ---
+   class: with-border
+   ---
+   ```
```
Comment (dhruvkb) [inline documentation/api/guides/deploy.md:20]: That's a very verbose way to add borders 😆. Not ideal.

File: documentation/api/guides/deploy.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+1. Release the app via
+   [GitHub workflow](https://github.com/WordPress/openverse/actions/workflows/release-app.yml).
+   Click the "Run workflow" button, choose "api" from the dropdown, and supply
+   the `sha` identified in step 1.
+1. That's it. The api will be deployed. You can monitor the deployment in the
```
Comment (dhruvkb) [inline documentation/api/guides/deploy.md:26]:
```suggestion
1. That's it. The API will be deployed. You can monitor the deployment in the
```

File: documentation/api/guides/deploy.md [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+```{tip}
+For more information on how deployments work, please see the [general deployment guide](/general/deployment.md).
+```
+
+1. Record and verify the commit `sha` of the production site at
+   [https://api.openverse.engineering/v1](https://api.openverse.engineering/v1).
+1. Visit
+   [https://api-staging.openverse.engineering/version](https://api-staging.openverse.engineering/version)
+   and
+   [the api package directory](https://github.com/wordpress/openverse/pkgs/container/openverse-api).
```
Comment (dhruvkb) [inline documentation/api/guides/deploy.md:12]:
```suggestion
   [the API Docker image](https://github.com/wordpress/openverse/pkgs/container/openverse-api).
```

**Review by dhruvkb — APPROVED** [review]
Content looks all good to me, based on what I observed @AetherUnbound and @krysal do in the catalog and ingestion server deploys respectively. I pointed out some small nits but didn't point out all occurrences of each change.

Approving to unblock as it is very helpful.

File: documentation/api/guides/deploy.md [hunk 8]
```diff
@@ -0,0 +1,42 @@
+# Deployment runbook
+
+```{tip}
+For more information on how deployments work, please see the [general deployment guide](/general/deployment.md).
+```
+
+1. Record and verify the commit `sha` of the production site at
+   [https://api.openverse.engineering/v1](https://api.openverse.engineering/v1).
```
[the remark this answers is not in the document]
Comment (sarayourfriend) [reply documentation/api/guides/deploy.md:8]: That applies to all the guides, BTW :slightly_smiling_face:

**Review by sarayourfriend — CHANGES_REQUESTED** [review]
LGTM! Very clear based on my understanding of these deployments having done them many times recently.

Dhruv's comment to undo the "check" variable change and a request to either clarify why the SHA is being saved or simply recommend using the changelog to derive the correct `rel-` tag to roll back to are my only blockers.

**Comment by sarayourfriend** [comment]
Converting to a draft to experiment with the PR reminder bot ping avoidance approach we've discussed elsewhere :slightly_smiling_face:

File: documentation/api/guides/deploy.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (zackkrida) [reply documentation/api/guides/deploy.md]: I removed those SHA recording steps and referenced finding the latest app version in the changelogs, instead.

File: documentation/api/guides/deploy.md (hunk 5 above)
Comment (zackkrida) [reply documentation/api/guides/deploy.md]: Copied this from one of my previous PRs, forgetting the `` markdown syntax works here. Fixed.

File: documentation/ingestion_server/guides/deploy.md [hunk 9]
```diff
@@ -0,0 +1,42 @@
+# Deployment runbook
+
+## Setup
+
+1. Check [airflow](https://airflow.openverse.engineering/home?tags=data_refresh)
+   to make sure a data refresh isn't occurring.
+2. Release the app via
+   [GitHub workflow](https://github.com/WordPress/openverse/actions/workflows/release-app.yml).
+   Click the "Run workflow" button, choose "ingestion server" from the dropdown,
+   and supply the SHA identified in step 1.
```
Comment (krysal) [inline documentation/ingestion_server/guides/deploy.md:10]: The SHa The SHA step is missing here.

```suggestion
1. Check [airflow](https://airflow.openverse.engineering/home?tags=data_refresh)
   to make sure a data refresh isn't occurring.
2. Visit the Ingestion Server Docker image page and copy the SHA of the one tagged with `latest`.
3. Release the app via
   [GitHub workflow](https://github.com/WordPress/openverse/actions/workflows/release-app.yml).
   Click the "Run workflow" button, choose "ingestion server" from the dropdown,
   and supply the SHA identified in step 2.
```

Comment (zackkrida) [reply documentation/ingestion_server/guides/deploy.md]: resolved in 9d35dcab3

**Review by sarayourfriend — APPROVED** [review]

**Pull request merged**